allennlp_models/rc/models/bidaf.py

```diff
@@ -275,8 +275,17 @@ def forward(  # type: ignore
         span_end_input = self._dropout(torch.cat([final_merged_passage, encoded_span_end], dim=-1))
         span_end_logits = self._span_end_predictor(span_end_input).squeeze(-1)
         span_end_probs = util.masked_softmax(span_end_logits, passage_mask)
-        span_start_logits = util.replace_masked_values(span_start_logits, passage_mask, -1e7)
-        span_end_logits = util.replace_masked_values(span_end_logits, passage_mask, -1e7)
+
+        # Replace the masked values with a very negative constant.
+        # We have to be careful here because if we're training with half/mixed precision floats,
+        # -1e7 would result in an overflow. Hence we fall back to the min FP16 value.
+        very_negative_constant = max(torch.finfo(span_start_logits.dtype).min, -1e7)
```

Comment:
I'm pretty sure we have a utility function for this. Looking it up...

Reply:
These: https://github.com/allenai/allennlp/blob/33a49454db239bfe3de6928229c405da056a8bf4/allennlp/nn/util.py#L1859-L1899.